library/systemd/src/lib/yast2/service_configuration.rb

```diff
@@ -0,0 +1,194 @@
+    #   services:
+    #
+    #   - `:active` when all services are active
+    #   - `:inactive` when all services are inactive
+    #   - `:inconsistent` when part of services is active and part not.
+    #       Can happen only when configuration handle multiple services
+    #   - `:unknown` when read of current status failed.
+    def status
+      read unless @status
+
+      @status
+    end
+
+    # returns currently set target action. If it is not yet set it returns
+    # `:nothing`.
+    # @return [:start, :stop, :restart, :nothing] action for all services:
+    #
+    #   - `:start` to start all services. If service is already active, do nothing.
+    #   - `:stop`  to stop all services. If service already is inactive, do nothing.
+    #   - `:restart` restart all services. If service is inactive, it is started.
+    #   - `:reload` reload all services that support it and restart that does not
+    #       support it. If service is inactive, it is started.
+    #   - `:nothing` do not touch anything.
+    def action
+      return :nothing unless @action
+
+      @action
+    end
+
+    ACTIONS = [ :start, :stop, :reload, :restart, :nothing].freeze
+    # sets target action for services
+    # @param action[:start, :stop, :restart, :reload, :nothing] for possible values and
+    #  its explanation please see return value of {target_action}
+    def action= (action)
+      if !ACTIONS.include?(action)
+        raise ArgumentError, "Invalid parameter #{action.inspect}"
+      end
+
+      @action = action
+    end
+
+    # returns currently set autostart configuration. If it is not yet set it
+    # calls read and return current system state.
+    # @return [:on_boot, :on_demand, :manual, :inconsistent, :unknown] autostart
+    #   configuration:
+    #
+    #   - `:on_boot` all services start during boot.
+    #   - `:on_demand`  all sockets associated with services is enabled and
+    #     for services that does not have socket, it starts on boot.
+    #   - `:manual` all services and all its associated sockets is disabled.
+    #   - `:inconsistent` mixture of states
+    #   - `:unknown` when no state is set and read call failed.
+    def autostart
+      read unless @autostart
+
+      @autostart
+    end
+
+    AUTOSTART_OPTIONS = [:on_boot, :on_demand, :manual, :inconsistent]
+    # sets autostart configuration.
+    # @param [:on_boot, :on_demand, :manual, :inconsistent] autostart
+    # configuratio. For explanation please see {autostart} when
+    # `:inconsistent` means keep it as it is now.
+    def autostart=(configuration)
+      if !AUTOSTART_OPTIONS.include?(configuration)
+        raise ArgumentError, "Invalid parameter #{configuration.inspect}"
+      end
+
+      @autostart = configuration
+    end
+
+    # returns true if any service support reload
+    def support_reload?
+      # TODO: implement it
+      true
+    end
+
+    # returns true if any service has socket start
+    def support_on_demand?
```

Comment:
And what if some of the services does not support loading on demand? Can we safely assume that on demand is supported for the whole group?

Reply:
well, for me it make sense in such case to start socket for services that support it and start on boot rest, so it will work. But in general I do not expect it to be common case. Probably only one I know now is iscsi client.